funsor/jax/distributions.py

```diff
@@ -215,6 +215,13 @@ def deltadist_to_data(funsor_dist, name_to_dim=None):
 # Converting PyTorch Distributions to funsors
 ###############################################
 
+dist.Independent.has_rsample = property(lambda self: self.base_dist.has_rsample)
+dist.Independent.rsample = dist.Independent.sample
+dist.MaskedDistribution.has_rsample = property(lambda self: self.base_dist.has_rsample)
+dist.MaskedDistribution.rsample = dist.MaskedDistribution.sample
+dist.TransformedDistribution.has_rsample = property(lambda self: self.base_dist.has_rsample)
+dist.TransformedDistribution.rsample = dist.TransformedDistribution.sample
```

Comment:
Can you add a TODO pointing to a NumPyro issue to fix this bug, so we can delete this workaround once the bug is fixed? cc @fehiepsi

Reply:
@neerajprad Should we add those new attributes to NumPyro distributions? We can make default behaviors for them so that there would be only a few changes in the code.

Reply:
Sounds good. In numpyro, for distributions that have reparametrized samplers available both will be the same so we can just add a default `Distribution.rsample` method which delegates to `sample` and throws a `NotImplemented` error when not available.